EpiAware/src/EpiObsModels/modifiers/TransformObservationModel.jl

```diff
@@ -0,0 +1,56 @@
+@doc raw"
+The `TransformObservationModel` struct represents an observation model that applies a transformation function to the expected observations before passing them to the underlying observation model.
+
+## Fields
+- `model::M`: The underlying observation model.
+- `transform::F`: The transformation function applied to the expected observations.
+
+## Constructors
+- `TransformObservationModel(model::M, transform::F = x -> log1pexp.(x)) where {M <: AbstractTuringObservationModel, F <: Function}`: Constructs a `TransformObservationModel` instance with the specified observation model and a default transformation function.
```

Comment:
It turns out that the `softplus` alias for `log1pexp` exists so probably marginally better to write that for readability

Reply:
In doc strings I mean. I'm very agnostic on this in the src code

Reply:
Do you think so? I wondere if log1pexp is better as its the explicit math and if our target users don't know what the softplus is that might be clearer? I don't feel very strongly either way

Reply:
I lean towards believing that expressions common in machine learning have a higher chance of being understood, but I don't really have strong evidence for that

Reply:
fair enough if that is our audience but its really epis and I think they probably won't have come across it. If you don't feel that strongly I think I would prefer to leave as is but what about adding something to the doc string to make it clear it is softplus?

Reply:
I added a very little signposting. Do you think that is enough?

Reply:
Yeah I think so.